test/Serilog.Tests/Settings/KeyValuePairSettingsTests.cs

```diff
@@ -152,5 +153,89 @@ public void TestMinimumLevelOverrides()
             Assert.NotNull(evt);
         }
 
+        [Theory]
+        [InlineData("Debug", LogEventLevel.Debug, "proper parsing")]
+        [InlineData("Information", LogEventLevel.Information, "proper parsing")]
+        [InlineData("", LogEventLevel.Information, "default value")]
+        public void LoggingLevelSwitchIsInstantiatedFromLogLevelAsString3(string paramValue, LogEventLevel expectedSwitchInitialLevel, string reason)
+        {
+            object actual = null;
+            var didSucceed = KeyValuePairSettings.TryInstantiate(typeof(LoggingLevelSwitch), paramValue, out actual);
+
+            Assert.True(didSucceed, "TryInstantiate should successfully instantiate a LoggingLevelSwitch");
+            Assert.NotNull(actual);
+            Assert.IsType<LoggingLevelSwitch>(actual);
+            Assert.Equal(expectedSwitchInitialLevel, ((LoggingLevelSwitch)actual).MinimumLevel);
+        }
+
+        [Fact]
+        public void LoggingLevelSwitchIsConfigured()
+        {
+            var settings = new Dictionary<string, string>
+            {
+                ["level-switch:Switch1"] = "Information",
+                ["minimum-level:controlled-by"] = "Switch1",
+            };
+
+            LogEvent evt = null;
+
+            var log = new LoggerConfiguration()
+                .ReadFrom.KeyValuePairs(settings)
+                .WriteTo.Sink(new DelegatingSink(e => evt = e))
+                .CreateLogger();
+
+            // TODO : maybe add Logger.LevelSwitch as an internal readonly property to avoid reflection here ? 
```

Comment:
I'm using reflection to access a private field `_levelSwitch` of class `Logger` .... 
I'm not super happy about that, but 
- I believe accessing that member is the only way to test that the Level switch is really applied (I have no "pointer" to the actual object because it was defined from config....)
- I was not sure whether it was ok to make the field `internal` to make it visible from the test (or insite an `internal` get-only property)

Reply:
Could we just infer the level by testing with a couple of different events? May be a bit clunky, but should stay stable over the long term via the public API.

Reply:
yes, that would prove that the initial value for the "level switch" was indeed passed, but not that it was assigned .... but I guess it is covered by the full test-case (passing the switch also to a sink)